site/app/controllers/UserProfileController.php

```diff
@@ -158,6 +158,33 @@ public function changeUserName() {
         }
     }
 
+    /**
+     * @Route("/user_profile/update_last_initial_format", methods={"POST"})
+     * @return JsonResponse
+     */
+    public function updateLastInitialFormat() {
+        $user = $this->core->getUser();
+        if (isset($_POST['format'])) {
+            $newVal = intval($_POST['format']);
+            // Handle the case where intval returns zero due to an invalid string
+            if ($newVal !== 0 || $_POST['format'] === '0' || $_POST['format'] === 0) {
+                try {
+                    $user->setLastInitialFormat($newVal);
+                    $user->setUserUpdated(true);
+                    $this->core->getQueries()->updateUser($user);
+                    return JsonResponse::getSuccessResponse([
+                        'message' => "Last initial format successfully updated!",
+                        'format' => $user->getLastInitialFormat(),
+                        'display_format' => $user->getDisplayLastInitialFormat(),
+                        'new_abbreviated_name' => $user->getDisplayAbbreviatedName()
+                    ]);
+                } finally {
+                }
```

Comment:
Why do we need an empty `finally` block here?

Reply:
I'm a little confused about what you're trying to do here.  Would it not be exactly equivalent to remove the `try-finally` stuff entirely?  If you don't catch an exception, it's going to fail either way.

Reply:
I think the best way to do that would be to catch the specific exception you're expecting and return an error if so.  As it is now, the try-finally isn't doing anything to catch exceptions, so it will fail if an exception occurs.